compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -1352,6 +1497,60 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
             return Ok(None);
```

Comment:
I think this should be gated in `probe_inherent_assoc_shared`. I don't want to support `T::INHERENT_CONST` in the type system until `inherent_associated_types` is stable. At the very least would need to sit down and properly figure out if all the problems `inherent_associated_types` faces that are blocking stabilization are irrelevant for consts which I don't want to do rn.

Reply:
Hmm, I don't know if this is possible because afaict, `T::INHERENT_CONST` is already supported in the type system on stable:

```rust
struct S;

impl S {
    const N: usize = 42;
}

fn foo() -> [(); S::N] { todo!() }
```

Unless I misunderstood?

Reply:
Yeah it would be a breaking change but this is all feature gated by `min_generic_const_args` so it won't affect stable, it's only a problem for actual stabilization and we can see about stabilizing `inherent_associated_types` first